src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAnAotApp.cs

```diff
@@ -524,6 +524,42 @@ public void Only_Aot_warnings_are_produced_if_EnableAotAnalyzer_is_set(string ta
                 .And.NotHaveStdOutContaining("warning IL3002");
         }
 
+        [RequiresMSBuildVersionTheory("17.0.0.32901")]
+        [InlineData(ToolsetInfo.CurrentTargetFramework)]
+        public void IsAotCompatible_implies_enable_analyzers(string targetFramework)
+        {
+            var projectName = "WarningAppWithAotAnalyzer";
+            var testProject = CreateTestProjectWithAnalysisWarnings(targetFramework, projectName, true);
+            testProject.AdditionalProperties["IsAotCompatible"] = "true";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+            buildCommand
+                .Execute(RuntimeIdentifier)
+                .Should().Pass()
+                .And.HaveStdOutContaining("warning IL3050")
+                .And.HaveStdOutContaining("warning IL3056")
+                .And.HaveStdOutContaining("warning IL2026")
+                .And.HaveStdOutContaining("warning IL3002");
+
+            var outputDirectory = buildCommand.GetOutputDirectory(targetFramework).FullName;
+            var assemblyPath = Path.Combine(outputDirectory, $"{projectName}.dll");
+
+            // injects the IsTrimmable attribute
+            AssemblyInfo.Get(assemblyPath)["AssemblyMetadataAttribute"].Should().Be("IsTrimmable:True");
```

Comment:
I'm not sure why, but this is failing in CI:

```
   Microsoft.NET.Publish.Tests.GivenThatWeWantToPublishAnAotApp.IsAotCompatible_implies_enable_analyzers(targetFramework: "net8.0") [FAIL]
      System.IO.FileNotFoundException : Could not find file 'C:\h\w\A4E408F0\t\dotnetSdkTests\v4xmjh5h.pae\IsAotCompatib---48D17CD7\WarningAppWithAotAnalyzer\bin\Debug\net8.0\WarningAppWithAotAnalyzer.dll'.
      Stack Trace:
           at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
           at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
           at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
           at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
           at System.IO.File.OpenRead(String path)
        /_/src/Tests/Microsoft.NET.TestFramework/AssemblyInfo.cs(57,0): at Microsoft.NET.TestFramework.AssemblyInfo.Get(String assemblyPath)
        /_/src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAnAotApp.cs(549,0): at Microsoft.NET.Publish.Tests.GivenThatWeWantToPublishAnAotApp.IsAotCompatible_implies_enable_analyzers(String targetFramework)
           at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
           at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
      Output:
        > C:\h\w\A4E408F0\p\d\dotnet.exe msbuild /t:Build C:\h\w\A4E408F0\t\dotnetSdkTests\v4xmjh5h.pae\IsAotCompatib---48D17CD7\WarningAppWithAotAnalyzer\WarningAppWithAotAnalyzer.csproj /restore /p:RuntimeIdentifier=win10-x64
```

Reply:
Yeah, trying to debug. Not having great success with running the test locally.